source/_drafts/Inband-Captions-Support-with-VHS.md

```diff
@@ -0,0 +1,93 @@
+---
+title: In-band Captions Support with videojs-http-streaming
+author:
+  name: Lahiru Dayananda
+  github: ldayananda
+tags:
+  - vhs
+  - videojs-http-streaming
+  - playback
+  - captions
+  - cea-608
+  - 608
+---
+
+With the release of [videojs-http-streaming](https://blog.videojs.com/introducing-video-js-http-streaming-vhs/) v1.2.0 on July 16th 2018, Video.js has built-in support for CEA/CTA-608 captions carried in FMP4 segments. This means that closed captions are automatically parsed out and made available to Video.js players for MPEG-DASH content and HLS streams using FMP4 segments.
+
+Caption Parsing is handled by the [mux.js]() library and interacts with videojs-http-streaming to feed parsed captions back to Video.js.
+
+## Usage
+
+Create a CaptionParser:
+
+```js
+  import { CaptionParser } from 'mux.js/lib/mp4';
+
+  const captionParser = new CaptionParser();
+
+  // initalize the CaptionParser to ensure that it is ready for data
+  if (!captionParser.isInitialized()) {
+    captionParser.init();
+  }
+```
+
+When working with HLS and MPEG-DASH with fmp4 segments, it's likely that not all the information needed to parse out captions are included in the media segments themselves, and metadata from the `init` segment needs to be passed to the CaptionParser. For this reason, the ids of video `trak`s and timescales defined in the `init` segment should be passed into the CaptionParser.
```

Comment:
`trak` -> `track`?

Reply:
I was actually referring to `traf -> tfhd -> trackId`, so yeah, `track` is probably better

Reply:
ah, yeah, I think seeing `trak` would be a bit confusing unless it was brought up beforehand somewhere.